test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -1012,98 +976,167 @@ def revert_pbsconf(self):
+                if new_pbsconf["PBS_START_COMM"] != "0":
+                    new_pbsconf["PBS_START_COMM"] = "0"
+                    cmds_to_exec.append(["comm", "stop"])
+            else:
+                if (new_pbsconf["PBS_START_COMM"] != "1"):
+                    new_pbsconf["PBS_START_COMM"] = "1"
+                    cmds_to_exec.append(["comm", "start"])
+                    dmns_to_start += 1
+
+            if dmns_to_start == 4:
+                # If all daemons need to be started again, just restart PBS
+                # instead of making PTL start each of them one at a time
+                restart_pbs = True
 
             # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
-            if default_vals["PBS_CORE_LIMIT"] != "unlimited":
-                default_vals["PBS_CORE_LIMIT"] = "unlimited"
-                start_comm = True
-            if default_vals["PBS_SERVER"] != primary_server.hostname:
-                default_vals["PBS_SERVER"] = primary_server.hostname
-                start_comm = True
-            if "PBS_SCP" not in default_vals:
-                scppath = self.du.which(comm.hostname, "scp")
+            if new_pbsconf["PBS_CORE_LIMIT"] != "unlimited":
+                new_pbsconf["PBS_CORE_LIMIT"] = "unlimited"
+                restart_pbs = True
+            if new_pbsconf["PBS_SERVER"] != primary_server.hostname:
+                new_pbsconf["PBS_SERVER"] = primary_server.hostname
+                restart_pbs = True
+            if "PBS_SCP" not in new_pbsconf:
+                scppath = self.du.which(server.hostname, "scp")
                 if scppath != "scp":
-                    default_vals["PBS_SCP"] = scppath
-                    start_comm = True
+                    new_pbsconf["PBS_SCP"] = scppath
+                    restart_pbs = True
 
             # Check if existing pbs.conf has more/less entries than the
             # default list
-            if len(pbs_conf_val) != len(default_vals):
-                start_comm = True
+            if len(pbs_conf_val) != len(new_pbsconf):
+                restart_pbs = True
+
+            if restart_pbs or dmns_to_start > 0:
+                # Write out the new pbs.conf file
+                self.du.set_pbs_config(server.hostname, confs=new_pbsconf,
+                                       append=False)
+                server.pbs_conf = new_pbsconf
+
+                if restart_pbs:
+                    # Restart all
+                    server.pi.restart(server.hostname)
+                else:
+                    for initcmd in cmds_to_exec:
+                        # start/stop the particular daemon
+                        server.pi.initd(server.hostname, initcmd[1],
+                                        daemon=initcmd[0])
+
+    def revert_pbsconf(self):
+        """
+        Revert contents of the pbs.conf file
+        Also start/stop the appropriate daemons
+        """
+        primary_server = self.servers.values()[0]
```

Comment:
Is this the primary server, or is self.server the primary server?

Reply:
If you look at init_servers(), self.server is actually set to servers.values()[0]:
        if cls.servers:
            cls.server = cls.servers.values()[0]

But I think it's probably safer to use self.server as the primary server, so I'll make that change, thanks.